ambry-network/src/main/java/com/github/ambry/network/http2/Http2BlockingChannelStreamChannelInitializer.java

```diff
@@ -25,22 +26,25 @@
  * A ChannelInitializer used to setup stream channel pipeline once stream is created in {@link MultiplexedChannelRecord}.
  */
 public class Http2BlockingChannelStreamChannelInitializer extends ChannelInitializer {
-  private final int http2MaxContentLength;
+  private final Http2ClientConfig http2ClientConfig;
   private static final Http2StreamFrameToHttpObjectCodec http2StreamFrameToHttpObjectCodec =
       new Http2StreamFrameToHttpObjectCodec(false);
   private static final Http2BlockingChannelResponseHandler http2BlockingChannelResponseHandler =
       new Http2BlockingChannelResponseHandler();
-  private static final AmbrySendToHttp2Adaptor ambrySendToHttp2Adaptor = new AmbrySendToHttp2Adaptor(false);
+  private static AmbrySendToHttp2Adaptor ambrySendToHttp2Adaptor = null;
 
-  Http2BlockingChannelStreamChannelInitializer(int http2MaxContentLength) {
-    this.http2MaxContentLength = http2MaxContentLength;
+  Http2BlockingChannelStreamChannelInitializer(Http2ClientConfig http2ClientConfig) {
+    this.http2ClientConfig = http2ClientConfig;
+    if (ambrySendToHttp2Adaptor == null) {
+      ambrySendToHttp2Adaptor = new AmbrySendToHttp2Adaptor(false, http2ClientConfig.http2FrameMaxSize);
+    }
```

Comment:
is this `if` statement necessary? i think we can just create `AmbrySendToHttp2Adaptor` without this if statement.

Reply:
without the if, we need to new an AmbrySendToHttp2Adaptor object, but AmbrySendToHttp2Adaptor is stateless. It can shared.

Reply:
Can we instead pass in the single `AmbrySendToHttp2Adaptor` to the constructor?

Reply:
It seems like we only have one instance of `Http2BlockingChannelStreamChannelInitializer` per `Http2BlockingChannelPool` anyways, so maybe you could keep this as a member variable instead of making it static?

Reply:
you are right. A member can solve this cleanly.